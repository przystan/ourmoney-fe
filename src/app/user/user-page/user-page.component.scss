:host {
  display: grid;
  grid-template-columns: 3fr 2fr;
  width: 100%;
  height: 100%;
}

.forms {
  display: flex;
  justify-content: center;
  align-items: center;
}
