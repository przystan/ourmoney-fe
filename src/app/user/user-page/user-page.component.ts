import { Component, ChangeDetectionStrategy } from '@angular/core';

@Component({
  selector: 'app-user-page',
  template: `
    <app-user-welcome></app-user-welcome>
    <div class="forms">
      <router-outlet></router-outlet>
    </div>
  `,
  styleUrls: ['./user-page.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class UserPageComponent {}
