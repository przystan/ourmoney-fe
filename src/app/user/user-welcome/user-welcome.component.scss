:host {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
  width: 100%;
  background-image:
    linear-gradient(
      var(--primary-background-color),
      var(--secondary-background-color)
    ),
    url('/assets/images/user-background.jpg');
  background-size: cover;
}

mat-icon {
  width: 150pt;
  height: 130pt;
  font-size: 150pt;
  color: white;
  opacity: 0.8;
}

.title {
  font-weight: lighter;
  opacity: 0.9;
  font-size: 4rem;
  color: white;
  font-family: 'Barlow Semi Condensed', sans-serif;
}

.description {
  margin-top: 2rem;
  color: white;
  opacity: 0.6;
}
