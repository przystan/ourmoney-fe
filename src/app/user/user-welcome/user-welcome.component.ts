import { Component, ChangeDetectionStrategy } from '@angular/core';

@Component({
  selector: 'app-user-welcome',
  template: `
    <mat-icon>account_balance_wallet</mat-icon>
    <div class="title">our money</div>
    <div class="description">
      Home budget management application
    </div>
  `,
  styleUrls: ['./user-welcome.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class UserWelcomeComponent {}
