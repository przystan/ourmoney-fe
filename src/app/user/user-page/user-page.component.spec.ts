import { MockComponent } from 'ng-mocks';
import { RouterOutlet } from '@angular/router';
import { Spectator, createComponentFactory } from '@ngneat/spectator';
import { UserPageComponent } from './user-page.component';
import { UserWelcomeComponent } from '../user-welcome/user-welcome.component';

describe('UserPageComponent', () => {
  let spectator: Spectator<UserPageComponent>;

  const createComponent = createComponentFactory({
    component: UserPageComponent,
    declarations: [
      MockComponent(RouterOutlet),
      MockComponent(UserWelcomeComponent)
    ]
  });

  beforeEach(() => {
    spectator = createComponent();
  });

  it('should match snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });
});
