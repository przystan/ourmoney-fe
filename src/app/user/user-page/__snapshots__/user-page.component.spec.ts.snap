// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPageComponent should match snapshot 1`] = `
<app-user-page>
  <app-user-welcome /><div
    class="forms"
  >
    <router-outlet />
  </div>
</app-user-page>
`;
